cloudbuild.yaml

```diff
@@ -19,7 +19,50 @@ steps:
       echo "Write .env.cloud_build to disk, to provide consistent project configuration (no secrets) in subsequent steps"
       deploy/make_cloud_build_env_file.sh "${BRANCH_NAME}" "${COMMIT_SHA}"
 
-  - id: "Build image"
+  - id: "Build server and postgres images for testing"
+    name: "gcr.io/cloud-builders/docker"
+    script: |
+      #!/usr/bin/env bash
+      set -o errexit
+      set -o pipefail
+      set -o nounset
+      
+      echo "Get cloud build project configuration values"
+      source .env.cloud_build
+
+      echo "Build new server and postgres db images for testing"
+      docker compose --env-file .env.cloud_build up --build -d 
```

Comment:
I see `up` is used here to start the container used in `"Run tests and produce coverage report"` and eventually killed in `"Tear down from tests"`. I'd prefer the `up` and `down` to happen within the same Cloud Build step (meaning along with running the tests, although building the composed image could still be separate probably). Let me know if there was some particular reason to split things in to multiple steps!

The reason I want the life time of the container to be contained to one step is because I expect, as some point, we'll start running different cloud build steps in parallel. I think things will get confusing/possibly buggy at that point if steps aren't self contained. Maybe that's not the right way to think about Cloud Build steps though, I'm still not fully comfortable with them in some ways.

Reply:
Yep - this makes sense!  Had them split out before as they were sandwiching the "upload to codecov step"(that also downloaded the uploader).  Will make this change.

Reply:
Tests (including build and tear down have been consolidated into one step).

[Removed extra steps in build - now the build images, run tests and te…](https://github.com/PHACDataHub/cpho-phase2/commit/936248c9616e6353f76c34a11dc60f95b7d4f7df)

[Simplified title for Run tests step of cloudbuild.](https://github.com/PHACDataHub/cpho-phase2/commit/179ede0e5af3835f74e01584ccccddea31ce05a1)